src/main/java/com/synopsys/integration/alert/channel/jira/util/JiraTransitionHelper.java

```diff
@@ -0,0 +1,138 @@
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.synopsys.integration.alert.channel.jira.util;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.synopsys.integration.alert.channel.jira.JiraIssueConfigValidator;
+import com.synopsys.integration.alert.channel.jira.descriptor.JiraDescriptor;
+import com.synopsys.integration.alert.channel.jira.exception.JiraMissingTransitionException;
+import com.synopsys.integration.alert.common.enumeration.ItemOperation;
+import com.synopsys.integration.exception.IntegrationException;
+import com.synopsys.integration.jira.common.cloud.builder.IssueRequestModelFieldsBuilder;
+import com.synopsys.integration.jira.common.cloud.model.components.IdComponent;
+import com.synopsys.integration.jira.common.cloud.model.components.StatusCategory;
+import com.synopsys.integration.jira.common.cloud.model.components.StatusDetailsComponent;
+import com.synopsys.integration.jira.common.cloud.model.components.TransitionComponent;
+import com.synopsys.integration.jira.common.cloud.model.request.IssueRequestModel;
+import com.synopsys.integration.jira.common.cloud.model.response.TransitionsResponseModel;
+import com.synopsys.integration.jira.common.cloud.rest.service.IssueService;
+
+public class JiraTransitionHelper {
+    public static final String TODO_STATUS_CATEGORY_KEY = "new";
+    public static final String DONE_STATUS_CATEGORY_KEY = "done";
+
+    private final Logger logger = LoggerFactory.getLogger(JiraTransitionHelper.class);
+    private IssueService issueService;
+
+    public JiraTransitionHelper(IssueService issueService) {
+        this.issueService = issueService;
+    }
+
+    public boolean doesTransitionToExpectedStatusCategory(TransitionComponent transition, String expectedStatusCategoryKey) throws IntegrationException {
+        StatusDetailsComponent statusDetails = transition.getTo();
+        StatusCategory statusCategory = statusDetails.getStatusCategory();
+        return StringUtils.equals(expectedStatusCategoryKey, statusCategory.getKey());
+    }
+
+    public boolean transitionIssueIfNecessary(String issueKey, JiraIssueConfigValidator.JiraIssueConfig jiraIssueConfig, ItemOperation operation) throws IntegrationException {
+        Optional<String> optionalTransitionKey = determineTransitionKey(operation);
```

Comment:
This method is used a single time and never uses the value of the passed optional. The method can be simplified to return a boolean.